Sources/AppStoreConnectCLI/Services/Operations/ReadPreReleaseVersionOperation.swift

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020 Itty Bitty Apps Pty Ltd
+
+import AppStoreConnect_Swift_SDK
+import Combine
+import Foundation
+
+struct ReadPreReleaseVersionOperation: APIOperation {
+
+    struct Options {
+        let filterAppId: String
+        let filterVersion: String
+    }
+
+    enum Error: LocalizedError {
+        case noVersionExists
+        case versionNotUnique
+
+        var errorDescription: String? {
+            switch self {
+            case .noVersionExists:
+                return "No prerelease version exists"
+            case .versionNotUnique:
+                return "More than 1 prerelease version returned"
+            }
+        }
+    }
+
+    typealias PreReleaseVersion = AppStoreConnect_Swift_SDK.PrereleaseVersion
+    typealias Relationships = [AppStoreConnect_Swift_SDK.PreReleaseVersionRelationship]?
+    typealias Output = (preReleaseVersion: PreReleaseVersion, relationships: Relationships)
+
+    private let options: Options
+
+    init(options: Options) {
+        self.options = options
+    }
+
+    func execute(with requestor: EndpointRequestor) -> AnyPublisher<Output, Swift.Error> {
+        var filters: [ListPrereleaseVersions.Filter] = []
+        filters += options.filterAppId.isEmpty ? [] : [.app([options.filterAppId])]
```

Comment:
Nice.

Reply:
Adam had suggested to write it this way in an earlier PR so I am just following that. It's quite nice and short.